services/analytics/object_details_test.go

```diff
@@ -293,3 +294,241 @@ func TestService_GetMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestService_GetLabels(t *testing.T) {
+	db := setup()
+	rm := models.NewReporter(db)
+	mm := models.NewMetrics(db)
+	t1, _ := time.Parse(time.RFC3339, "2019-01-01T00:00:00Z")
+	t2, _ := time.Parse(time.RFC3339, "2019-01-01T10:00:00Z")
+	want := qanpb.ObjectDetailsLabelsReply{}
+
+	type fields struct {
+		rm models.Reporter
+		mm models.Metrics
+	}
+	type args struct {
+		ctx context.Context
+		in  *qanpb.ObjectDetailsLabelsRequest
+	}
+	type testCase struct {
+		name    string
+		fields  fields
+		args    args
+		want    *qanpb.ObjectDetailsLabelsReply
+		wantErr error
+	}
+
+	tt := testCase{
+		"success",
+		fields{rm: rm, mm: mm},
+		args{
+			context.TODO(),
+			&qanpb.ObjectDetailsLabelsRequest{
+				PeriodStartFrom: &timestamp.Timestamp{Seconds: t1.Unix()},
+				PeriodStartTo:   &timestamp.Timestamp{Seconds: t2.Unix()},
+				GroupBy:         "queryid",
+				FilterBy:        "1D410B4BE5060972",
+			},
+		},
+		&want,
+		nil,
+	}
+
+	t.Run(tt.name, func(t *testing.T) {
+		s := &Service{
+			rm: tt.fields.rm,
+			mm: tt.fields.mm,
+		}
+		got, err := s.GetLabels(tt.args.ctx, tt.args.in)
+		if err != tt.wantErr {
```

Comment:
Could we also start using testify in a new code?
```go
require.Equal(t, tt.wantErr, err)
```

Reply:
👍